org.eclipse.jdt.core/formatter/org/eclipse/jdt/internal/formatter/CommentsPreparator.java

```diff
@@ -793,6 +796,165 @@ private void alignJavadocTag(List<Token> tagTokens, int paramNameAlign, int desc
+							disableFormattingExclusively(tokenIndex, tokenIndexLast + 1);
+						}
+						this.snippetForMarkdown = false;
+					} else {
+						disableFormattingExclusively(tokenIndex, tokenIndexLast + 1);
+					}
+					hasFormatted = true;
+					break;
+				}
+
+			}
+			if (!hasFormatted) {
+				disableFormattingExclusively(tokenIndex, tokenIndexLast + 1);
+			}
+		}
+	}
+
+	private void handleMarkdownList(List<Object> fragments, int nodeStartPosition, String source) {
+		Matcher matcher;
+		Deque<Integer> indentPerLevel = new ArrayDeque<>();
+		for (Object fragment : fragments) {
+			if (fragment instanceof TextElement textElement) {
+				String textContent = textElement.getText();
+				matcher = MARKDOWN_LIST_PATTERN.matcher(textContent);
+				if (matcher.find()) {
+					int startPos = matcher.start() + textElement.getStartPosition();
+					int bulletIndex = tokenStartingAt(startPos);
+					Token bulletToken = this.ctm.get(bulletIndex);
+					int slashPos = source.lastIndexOf('/', matcher.start()) + nodeStartPosition;
+
+					int bulletIndent = this.ctm.getLength(slashPos + 1, bulletToken.originalStart - 1, 0);
+					if (bulletIndent > 1 && (bulletIndent % 2) != 0) {
+						bulletIndent--;
+					}
+					if (bulletIndex != 1) {
+						bulletToken.breakBefore();
+					}
+					while (!indentPerLevel.isEmpty() && bulletIndent < indentPerLevel.peekLast()) {
+						indentPerLevel.removeLast();
+					}
+
+					if (indentPerLevel.isEmpty() || bulletIndent > indentPerLevel.peekLast()) {
+						indentPerLevel.addLast(bulletIndent);
+					}
+
+					int currentLevel = indentPerLevel.size();
+					int indentToSet = 2 * (currentLevel - 1);
+					bulletToken.setIndent(indentToSet);
+					bulletToken.spaceAfter();
+				}
+			}
+		}
+	}
+
+	private void handleMarkdownTable(List<Object> fragments) {
+		int tableStartIndex = -1;
+		int tableLastIndex = -1;
+		boolean columnUnderlineFound = false;
+		Matcher matcher;
+		for (Object fragment : fragments) {
+			if (fragment instanceof TextElement textElement) {
+				String textContent = textElement.getText();
+				matcher = MARKDOWN_TABLE_START.matcher(textContent);
+				if (tableStartIndex == -1 && matcher.find()) {
+					int startPos = matcher.start() + textElement.getStartPosition();
+					tableStartIndex = tokenStartingAt(startPos);
+				} else if (tableStartIndex != -1 && !columnUnderlineFound) {
+					boolean foundStart = textContent.contains("|-"); //$NON-NLS-1$
+					boolean foundEnd = textContent.contains("-|"); //$NON-NLS-1$
+					if (foundStart && foundEnd) {
+						columnUnderlineFound = true;
+					}
+				} else if (columnUnderlineFound && matcher.find()) {
+					matcher = MARKDOWN_TABLE_END.matcher(textContent);
+					matcher.find(); // find the last one
+					int startPos = matcher.start() + textElement.getStartPosition();
+					tableLastIndex = tokenStartingAt(startPos);
+				}
+			}
+		}
+		if (tableStartIndex != -1 && tableLastIndex != -1) {
```

Comment:
looks like this should be inside the loop, otherwise it will not handle more than one table per comment